mono/eglib/test/memory.c

```diff
@@ -30,8 +30,100 @@ test_memory_zero_size_allocations (void)
         return OK;
 }
 
+static RESULT
+test_align_signed_int (void)
+{
+	gssize orig_value = 67;
+	gssize result, exp_value;
+	gpointer orig_ptr = (gpointer)orig_value;
+	gpointer result_ptr, exp_ptr;
+
+	// test case #1
+        int align = 1;
+	result = ALIGN_TO(orig_value, align);
+	exp_value = orig_value;
+	if (result != exp_value)
+                return FAILED ("Expected value after aligned is %d, however, the actual value is %d", exp_value, result);
+
+	result = ALIGN_DOWN_TO(orig_value, align);
+	exp_value = orig_value;
+	if (result != exp_value)
+                return FAILED ("Expected value after aligned is %d, however, the actual value is %d", exp_value, result);
+
+	result_ptr = ALIGN_PTR_TO(orig_ptr,align);
+	exp_ptr = orig_ptr;
+	if (result_ptr != exp_ptr)
+                return FAILED ("Expected address after aligned is %p, however, the actual address is %p", exp_ptr, result_ptr);
+
+	// test case #2
+	align = 8;
+	result = ALIGN_TO(orig_value, align);
+	exp_value = 72;
+	if (result != exp_value)
+                return FAILED ("Expected value after aligned is %d, however, the actual value is %d", exp_value, result);
+
+	result_ptr = ALIGN_PTR_TO(orig_ptr,align);
+	exp_ptr= (gpointer)exp_value;
+	if (result_ptr != exp_ptr)
+                return FAILED ("Expected address after aligned is %p, however, the actual address is %p", exp_ptr, result_ptr);
+
+	result = ALIGN_DOWN_TO(orig_value, align);
+	exp_value = 64;
+	if (result != exp_value)
+                return FAILED ("Expected value after aligned is %d, however, the actual value is %d", exp_value, result);
+
+	return OK;
+}
+
+static RESULT
+test_align_unsigned_int (void)
+{
+	gssize orig_value = 67;
+	gssize result, exp_value;
+	gpointer orig_ptr = (gpointer)orig_value;
+        gpointer result_ptr, exp_ptr;
+
+	// test case #1
+        unsigned int align = 1;
```

Comment:
guint?

Reply:
Are g*** types the recommended ones over default C types?
I see people use both of them...
BTW, do we have these best practices documented anywhere?

Reply:
In general I think the policy is to avoid it for things like `char`, but for numerical types use the `guint32` form since it's shorter and unambiguous? @lambdageek can probably comment, but no, I don't think it's properly documented anywhere.

Reply:
TL;DR - if you know you definitely want a 32- or 64-bit number, use the C99 `int32_t`/`uint32_t`/`int64_t`/`uint64_t` types.  If you just want an `int` use an `int`.

The `gint32`/etc types are from before we had good C99 support from all the compilers we cared about.

I'm not totally convinced we need the `g` types for things where good types exist in the standard now.  The one exception seems to be `gboolean` which we're using pervasively and we're not using `bool` or `_Bool` from C99 anywhere - so in that one case we seem to prefer `gboolean`.

You may also come across `MonoBoolean` - that's used in structures and function signatures of functions that are called from C# - the C# `bool` type corresponds to `MonoBoolean`, not `gboolean`.

Reply:
gotcha

Reply:
`guint` is shorthand for `unsigned`; or consider `guint32`
`gushort` is shorthand for `unsigned short`; or consider `guint16`
`guchar` is shorthand for `unsigned char`; or consider `guint8`

The following are pointless:
 `gchar` just use `char`
 `gint` just use `int`, or consider `gint32`, but really `int` suffices
 `gshort` just use `short`, or consider `gint16`
 `glong` use `long` or really avoid altogether, it is too variably defined across platforms

`gint32` questionable, could be `int`; 16bit `int` is irrelevant
`gint8` could be `int8_t` but it is longer; etc.
`MonoBoolean` would be `bool` if using C++.

Reply:
Another wrinkle is `g` is not to be used in public headers, but `int64_t` can be.
I'd suggest really cleanup the code base mechanically and thoroughly and drive out the `g` types.